plugins/governance/resolver.go

```diff
@@ -109,36 +159,42 @@ func (r *BudgetResolver) EvaluateRequest(ctx *schemas.BifrostContext, evaluation
 	}
 
 	// 2. Check provider filtering
-	if !r.isProviderAllowed(vk, evaluationRequest.Provider) {
+	if !r.isProviderAllowed(vk, provider) {
 		return &EvaluationResult{
 			Decision:   DecisionProviderBlocked,
-			Reason:     fmt.Sprintf("Provider '%s' is not allowed for this virtual key", evaluationRequest.Provider),
+			Reason:     fmt.Sprintf("Provider '%s' is not allowed for this virtual key", provider),
 			VirtualKey: vk,
 		}
 	}
 
 	// 3. Check model filtering
-	if !r.isModelAllowed(vk, evaluationRequest.Provider, evaluationRequest.Model) {
+	if !r.isModelAllowed(vk, provider, model) {
 		return &EvaluationResult{
 			Decision:   DecisionModelBlocked,
-			Reason:     fmt.Sprintf("Model '%s' is not allowed for this virtual key", evaluationRequest.Model),
+			Reason:     fmt.Sprintf("Model '%s' is not allowed for this virtual key", model),
 			VirtualKey: vk,
 		}
 	}
 
-	// 4. Check rate limits hierarchy (Provider level first, then VK level)
-	if rateLimitResult := r.checkRateLimitHierarchy(ctx, vk, string(evaluationRequest.Provider), evaluationRequest.Model, evaluationRequest.RequestID); rateLimitResult != nil {
```

Comment:
lets have evaluationRequest here in checkRateLimitHierarchy as well so there's symmetry between this and checkBudgetHierarchy